### ISSUE TITLE
cmd/compile: error message for missing method on type with another type embedded talks about the embedded type instead

### ISSUE THREAD

**Issue by go101** [golang/go#47833]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.17 linux/amd64
</pre>

### Does this issue reproduce with the latest release?
Yes

### What did you do?

```Go
package main

type T0 struct {
	x int
}

func (T0) M0() {}

type T2 struct {
	T0
}

type Q *T2

var q Q

func main() {
	q.M0()
}
```

### What did you expect to see?

```
q.M0 undefined (type Q has no field or method M0)
```

### What did you see instead?

```
q.T0.M0 undefined (type T0 has no field or method M0, but does have M0)
```

**Comment by mknyszek** [comment]
Is this new in Go 1.17?

CC @griesemer

**Comment by cuonglm** [comment]
This is fixed at tip 8fff20ffebadfea2c5c3dd58f7007246f3408495

Actually, it was fixed when the compiler switched to use types2. With go1.17, you can do:

```
$ go1.17 tool compile -G=3 r.go
r.go:18:4: q.M0 undefined (type Q has no field or method M0)
```

**Comment by go101** [comment]
As this is not qualified for backporting, I will close it.
